Utilities/build-using-self

```diff
@@ -148,23 +159,31 @@ def main() -> None:
         call(
             shlex.split("swift package update"),
         )
+
         call(
-            shlex.split(f"swift build --configuration {args.config} {ignore}"),
+            shlex.split(f"swift build --configuration {args.config} --build-tests {extra_args}"),
         )
 
         if os.name != "nt": # turn off for Windows until we get the hang resolved
-            swift_testing_arg= "--enable-swift-testing" if args.enable_swift_testing else ""
-            xctest_arg= "--enable-xctest" if args.enable_swift_testing else ""
+            swift_testing_arg = "--enable-swift-testing" if args.enable_swift_testing else ""
+            xctest_arg = "--enable-xctest" if args.enable_swift_testing else ""
+            swift_test_executable = swiftpm_bin_dir / 'swift-test'
             call(
-                shlex.split(f"swift run swift-test --configuration {args.config} --parallel {swift_testing_arg} {xctest_arg} --scratch-path .test {ignore}"),
+                shlex.split(
+                    f"{swift_test_executable} --configuration {args.config} --parallel {swift_testing_arg} {xctest_arg} --scratch-path .test {extra_args}",
+                    posix=(os.name == "posix"), # must be set correctly, otherwhsie shlex.split("C:\\Foo\\bar") become ['CFoobar']
+                ),
             )
 
             integration_test_dir = REPO_ROOT_PATH / "IntegrationTests"
             call(
                 shlex.split(f"swift package --package-path {integration_test_dir} update"),
             )
             call(
-                shlex.split(f"swift run swift-test --package-path {integration_test_dir} --parallel {ignore}"),
+                shlex.split(
+                    f"{swift_test_executable} --package-path {integration_test_dir} --parallel {extra_args}",
+                    posix=(os.name == "posix"), # must be set correctly, otherwhsie shlex.split("C:\\Foo\\bar") become ['CFoobar']
```

Comment:
```suggestion
                    posix=(os.name == "posix"), # must be set correctly, otherwise shlex.split("C:\\Foo\\bar") becomes ['CFoobar']
```